internal/storage/storageutil/control_client.go

```diff
@@ -58,7 +59,8 @@ func CreateGRPCControlClient(ctx context.Context, clientOpts []option.ClientOpti
 		logger.Fatal("error setting direct path env var: %v", err)
 	}
 
-	controlClient, err = control.NewStorageControlClient(ctx, clientOpts...)
+	opts := append(clientOpts, internaloption.EnableDirectPath(true), internaloption.AllowNonDefaultServiceAccount(true), internaloption.EnableDirectPathXds())
```

Comment:
I think the internaloptions are only meant to be used inside google-cloud-go libraries. @codyoss @quartzmo is there a config we can set on the generator to get these into the storage/control GAPIC by default?

Reply:
The `internaloption` values in a gRPC GAPIC are set [here](https://github.com/googleapis/gapic-generator-go/blob/main/internal/gengapic/gengrpc.go#L169-L201) in a pretty much static operation. The only conditional is if the client  supports gRPC.

> internaloptions are only meant to be used inside google-cloud-go libraries

The reason for this afaik is primarily the instability of the options. Most should be considered EXPERIMENTAL, and may be removed/replaced at any time. The maintainers of `internaloption` will not be on the lookout for usages outside of the core Go client libraries projects such as gapic-generator-go, google-cloud-go, and google-api-go-client. Using them here is risky.

Reply:
+1, I think this is fine for now. I will file an internal FR to have a non-internal option available for enabling DP in a GAPIC (or allow service owners to force this option to be supplied via a generator config).